tests/typecheck/managers/querysets/test_as_manager.yml

```diff
@@ -146,6 +146,32 @@
                 class MyModel(models.Model):
                     objects = ManagerFromModelQuerySet
 
+-   case: handles_subclasses_of_queryset
+    main: |
+        from myapp.models import MyModel
+        reveal_type(MyModel.objects.example())  # N: Revealed type is "myapp.models.MyModel"
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from typing import TypeVar
+
+                from django.db import models
+
+                _CTE = TypeVar("_CTE", bound=models.Model)
+
+                class _MyModelQuerySet(models.QuerySet[_CTE]):
+
+                    def example(self) -> _CTE: ...
+
+                class MyModelQuerySet(_MyModelQuerySet["MyModel"]):
+                    ...
+
+                class MyModel(models.Model):
+                    objects = MyModelQuerySet.as_manager()
```

Comment:
I think there's a missing `Generic` on `_MyModelQuerySet` in this test case? It looks odd with:

```python
class MyModelQuerySet(_MyModelQuerySet["MyModel"]):
    ...
```

Consider how mypy acts on generic classes not touched by the plugin:

```python
from typing import Generic, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def m(self, arg: T) -> T:
        return arg
        
        
class B(A[int]): ...


class C(B[str]): ...  # main.py:13: error: "B" expects no type arguments, but 1 given  [type-arg]
```

https://mypy-play.net/?mypy=latest&python=3.11&gist=5fc71cabc2fe3dc45cf3d379e6919ca0

So, in general, something is off here in regards to arguments..

Ref: https://github.com/typeddjango/django-stubs/pull/1649#issuecomment-1681846078

Reply:
If you mean 
`class _MyModelQuerySet(models.QuerySet[_CTE]):` instead of `class _MyModelQuerySet(models.QuerySet[_CTE], Generic[_CTE)):` ? I am not sure to see the difference for the test.

I can Add the Generic if you think this is more understandable.

And in your example B, as you point out, is not generic. This is what I want for MyModelQuerySet, for not being generic :

MyModelQuerySet is a queryset of MyModel and nothing else.

Reply:
In both cases I have the same mypy output :
```
from typing import Generic, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def m(self, arg: T) -> T:
        return arg
class B(A[T]): ...
class C(B[int]): ...
reveal_type(C().m(3))
```
test1.py:14: note: Revealed type is "builtins.int"

```
from typing import Generic, TypeVar

T = TypeVar("T")

class A(Generic[T]):
    def m(self, arg: T) -> T:
        return arg

class B(A[T], Generic[T]): ...

class C(B[int]): ...


reveal_type(C().m(3))
```
test1.py:14: note: Revealed type is "builtins.int

Reply:
Thanks for clearing that up. It's me not being used to not using a `Generic`.